src/metabase/query_processor/middleware/metrics.clj

```diff
@@ -230,6 +231,18 @@
       :else
       expanded-stages)))
 
+(defn- adjust-metric-stages-counting-errors
+  "Call [[adjust-metric-stages]] and inc an error counter if any exception is thrown."
+  [query path expanded-stages]
+  (try (adjust-metric-stages query path expanded-stages)
+       (catch Throwable e
+         (prometheus/inc-and-throw! :metabase-query-processor/metrics-errors e))))
+
+(defn- match-one-metric
+  [query]
+  (lib.util.match/match-one query
+    [:metric _ _] &match))
```

Comment:
What about `references-metrics?` (and calling `boolean` on the result of `match-one`)?

Reply:
The return value is used in the `when-let` when checking to see if the metric replacement failed.

We could do without this function of course, but I wanted to make it clear that the `if-not` and `when-let` were checking for the same condition.